java/vector/src/main/java/org/apache/arrow/vector/ZeroVector.java

```diff
@@ -158,116 +74,22 @@ public TransferPair makeTransferPair(ValueVector target) {
     return defaultPair;
   }
 
-  @Override
-  public FieldReader getReader() {
-    return NullReader.INSTANCE;
-  }
-
-  @Override
-  public void initializeChildrenFromFields(List<Field> children) {
-    if (!children.isEmpty()) {
-      throw new IllegalArgumentException("Zero vector has no children");
+  private final TransferPair defaultPair = new TransferPair() {
+    @Override
+    public void transfer() {
     }
-  }
-
-  @Override
-  public List<FieldVector> getChildrenFromFields() {
-    return Collections.emptyList();
-  }
 
-  @Override
-  public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers) {
-    if (!ownBuffers.isEmpty()) {
-      throw new IllegalArgumentException("Zero vector has no buffers");
+    @Override
+    public void splitAndTransfer(int startIndex, int length) {
     }
-  }
-
-  @Override
-  public List<ArrowBuf> getFieldBuffers() {
-    return Collections.emptyList();
-  }
-
-  @Override
-  public List<BufferBacked> getFieldInnerVectors() {
-    return Collections.emptyList();
-  }
-
-  @Override
-  public long getValidityBufferAddress() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public long getDataBufferAddress() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public long getOffsetBufferAddress() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public ArrowBuf getValidityBuffer() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public ArrowBuf getDataBuffer() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public ArrowBuf getOffsetBuffer() {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public int getValueCount() {
-    return 0;
-  }
-
-  @Override
-  public void setValueCount(int valueCount) {
-  }
-
-  @Override
-  public Object getObject(int index) {
-    return null;
-  }
-
-  @Override
-  public int getNullCount() {
-    return 0;
-  }
-
-  @Override
-  public boolean isNull(int index) {
-    return false;
-  }
-
-  @Override
-  public int hashCode(int index) {
-    return 0;
-  }
 
-  @Override
-  public <OUT, IN> OUT accept(VectorVisitor<OUT, IN> visitor, IN value) {
-    return visitor.visit(this, value);
-  }
-
-  @Override
-  public void copyFrom(int fromIndex, int thisIndex, ValueVector from) {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public void copyFromSafe(int fromIndex, int thisIndex, ValueVector from) {
-    throw new UnsupportedOperationException();
-  }
+    @Override
+    public ValueVector getTo() {
+      return ZeroVector.this;
+    }
 
-  @Override
-  public String getName() {
-    return getField().getName();
-  }
+    @Override
+    public void copyValueSafe(int from, int to) {
```

Comment:
should be an error if from/to are greater then zero?

Reply:
Since ZeroVector has no values, any index is invalid include 0, so just do nothing in this method.

Reply:
I guess by that argument an exception should be thrown no matter what?

Reply:
but this seems to be existing behavior.